opm/simulators/flow/GenericCpGridVanguard.cpp

```diff
@@ -207,6 +208,34 @@ doLoadBalance_(const Dune::EdgeWeightMethod             edgeWeightsMethod,
                                  eclState1, parallelWells);
         }
 
+        // Add inactive wells to all ranks with connections (not solved, so OK even without distributed wells)
+        // @NOTE: Approach below is just for testing, there has to be a better way...
+        std::unordered_set<unsigned> cellOnRank;
+        const auto& global_cells = this->grid_->globalCell();
+        for (const auto cell : global_cells) cellOnRank.insert(cell);
+        const auto& comm = this->grid_->comm();
+        const auto nranks = comm.size();
+        const auto rank = comm.rank();
+        const auto inactive_well_names = schedule.getInactiveWellNamesAtEnd();
+        for (const auto& well_name : inactive_well_names) {
+            const auto& well = schedule.getWell(well_name, schedule.size()-1);
+            std::vector<int> well_on_rank(nranks, 0);
+            for (const auto& conn: well.getConnections()) {
+                if (cellOnRank.count(conn.global_index()) > 0) {
+                    well_on_rank[rank] = 1;
+                    break;
+                }
+            }
+            std::vector<int> well_on_rank_global(nranks, 0);
+            MPI_Allreduce(&well_on_rank[0], &well_on_rank_global[0], nranks, MPI_INT, MPI_MAX, this->grid_->comm());
```

Comment:
To support building without MPI, I would suggest to use`comm.max(&well_on_rank[0], nranks);`

Reply:
Good point, will update.